### DISCUSSION TITLE
Configurable currency rates for conversion

### DISCUSSION THREAD

**Discussion by ivanalejandro0** [antonmedv/numbr#10] [Ideas]
Have you considered making the currency rates configurable?

On the documentation it reads that the exchange rates are updated every five minutes, but what if you want/need to use a different rate?

My use case is for ARS, the argentinian peso.

In Argentina, there are different exchange rates that are used for different things (regular banking, investing, etc), and the prices vary a lot between them (different taxes, market prices, etc). And several rates are valid and useful at the same time, is not like you can choose one and that'll be the right one.

So if I want to make a conversion is not enough to write "$100 in ARS", because depending on the type of rate you use the difference would be of 100%, so I have to do some manual conversion (multiplication and/or division).

Maybe a user could define the conversion rate that want to use somewhere in the document?

p.s.: if you're interested on more details of exchange rates in Argentina I can elaborate. There's also some websites that show most used rates every day, like https://www.ambito.com/contenidos/dolar.html or https://www.cronista.com/MercadosOnline/dolar.html

**Comment by antonmedv** [comment]
This is a nice idea. We can do something like this:

```
Rate ARS extra = 1 ARS + 5%
$100 in ARS extra
```

**Reply by ivanalejandro0** [reply]
Would it be `extra` a new "suffix keyword"?
or would we be able to define any name for a new rate using `Rate`? ( and on this case the new rate is named "ARS extra")

Like, could I do this?

```
Rate ARS_A = 123
Rate ARS_B = 456
$100 in ARS_A
$100 in ARS_B
```

Just some extra info, values from different exchange rates can vary a lot and not necessarily related with each other. An example for today's rates, we have "blue" at 285, "official" at 146, "CCL" at 300, etc.

Also, is it possible to define an "absolute" value for a rate (just a number) and not in relation to the rate that's already loaded in Numbr? That would be extremely easy to deal with all the variations of exchanges when necessary.

**Reply by antonmedv** [reply]
Also, maybe like this: 

    RATE(CCL, 123)

**Reply by ivanalejandro0** [reply]
Yeah, something like `RATE(CCL, 123)` works for me :).

And my 2 cents on how to implement this feature, for you to consider...

On the documentation I don't see any other function-like keywords and I wonder if this will be the best syntax to choose for two reasons:
* simplicity for users, maybe something like "define rate ccl = 123" would be better?
* possible collision with future math keywords, I wonder if there are plans to implement something like what you mentioned on https://github.com/antonmedv/numbr/issues/11 ... I don't know if "rate()" is something that should be reserved or not.

**Comment by LautiLosio** [comment]
Came here for this exact feature request hahahahaha. I'm also Argentinean and those exchange rates are now more divergent than ever. 

The official rate is ≈362
The unregulated market is ≈715 !!

I hope we get this feature soon... 😁
